### ISSUE TITLE
master: jupyterlab - ERROR: Could not find a version that satisfies the requirement requests (from jupyterlab-server==1.2.0) (from versions: none)

### ISSUE THREAD

**Issue by InLaw** [DavHau/mach-nix#151]
```
let
  mach-nix = import (builtins.fetchGit {
    url = "https://github.com/DavHau/mach-nix.git";
    ref = "refs/heads/master" ;
    rev ="cd24fad1c91a28eb19fa0ee1a8a1c8455a623079";
  } ) { # nix-shell -p nix-prefetch-git --run ' nix-prefetch-git https://github.com/DavHau/mach-nix.git refs/heads/master '
    pkgs = import <nixpkgs> {};
  } ; 
  nixPkgs = mach-nix.nixpkgs;

  pyEnv = mach-nix.mkPython rec { 
    python = "python37";  
    requirements = '' 
    jupyterlab
    '';
     _.jupyterlab.buildInputs.add = ( with nixPkgs.python37Packages; [requests ]) ; 

    providers = { 
      jupyterlab = "nixpkgs" ;
    };
  };
 envShell = nixPkgs.mkShell rec {   
  buildInputs =     [     
    (pyEnv ) 
  ]++(with nixPkgs ; [
    bash
    which 
    ]  ) 
     ;
    shellHook = '' 
      # nix-instantiate --eval -E '(import <nixpkgs> {}).lib.version'
      QT_DEBUG_PLUGINS=1 

      echo ""
      echo "Your env is:"
      echo  ${pyEnv}
      echo ""

      echo "R path : "
      #echo $(which R )
      # ${pyEnv}/bin/python -c "import rpy2.robjects as robjects "

      jupyter lab --version
      jupyter lab 
    '';
};
in envShell
```

---

```
Executing pipInstallPhase
/build/jupyterlab_server-1.2.0/dist /build/jupyterlab_server-1.2.0
Processing ./jupyterlab_server-1.2.0-py3-none-any.whl
Requirement already satisfied: json5 in /nix/store/cvjy6xgg8d09ffa7vrwfgv7j1wxynqrm-python3.7-pyjson5-0.8.5/lib/python3.7/site-packages (from jupyterlab-server==1.2.0) (0.8.5)
Requirement already satisfied: notebook>=4.2.0 in /nix/store/q3ssacdh9gd58hhqwxhr53n459awdm4s-python3.7-notebook-6.1.4/lib/python3.7/site-packages (from jupyterlab-server==1.2.0) (6.1.4)
ERROR: Could not find a version that satisfies the requirement requests (from jupyterlab-server==1.2.0) (from versions: none)
ERROR: No matching distribution found for requests (from jupyterlab-server==1.2.0)
builder for '/nix/store/j9yz2aybxir1q5i06268ghsqzmic26cj-python3.7-jupyterlab_server-1.2.0.drv' failed with exit code 1
cannot build derivation '/nix/store/3a8b0pivz1vhag2j4v0c332jvb7nkz4x-python3-3.7.9-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/3a8b0pivz1vhag2j4v0c332jvb7nkz4x-python3-3.7.9-env.drv' failed

```

**Comment by DavHau** [comment]
I was running into this problem yesterday, too. This particular error is already fixed, but even on 4d995d7872a0985e86d03c4968cec0322e39b597, your expression triggers some collision.

FYI: `_.jupyterlab.buildInputs.add = ( with nixPkgs.python37Packages; [requests ]) ; ` is not a good idea. Manually mixing in python packages can introduce collisions, since mach-nix has less control about those dependencies.
ALways add python packages via `requirements` or `extra_pkgs`

**Comment by DavHau** [comment]
There was still something wrong with the nixpkgs inheritance logic. I fixed it and added a test for jupyterlab from nixpkgs.
